### ISSUE TITLE
ItemsAdder chat format (colors & emojis) doesn't get parsed properly in chat

### ISSUE THREAD

**Issue by AGuyNamedJens** [LOOHP/InteractiveChat#163]
The characters <r INPUT> for rainbow text does get parsed correctly using `/interactivechat parse <r t>`, but doesn't properly load in the chat... Why's that?

![image](https://github.com/LOOHP/InteractiveChat/assets/34610614/b21ebf77-ce0a-4bb8-b42a-ee5d598feb7c) - Chat resukt
![image](https://github.com/LOOHP/InteractiveChat/assets/34610614/946655d2-051c-42b2-a1a7-8d752555eb16) - Parsed result

**Comment by AGuyNamedJens** [comment]
> This command is used for testing IC placeholders only. Does it work properly in normal chat?

No, that's the point. It works on the discordSRV addon, as well as through console (commands/messages NOT sent by players). Only when players chat it doesn't parse them properly

**Comment by AGuyNamedJens** [comment]
@LOOHP any ideas?

**Comment by AGuyNamedJens** [comment]
> Try every priority setting in `Settings.EventPriorities.Chat` one by one, see whether any of them makes a different.

Doesn't seem to have any effect, do i restart the server each time?

**Comment by AGuyNamedJens** [comment]
Disabling both of these: ```  # This option is only effective on Minecraft 1.19.1+
  ForceUnsignedChatPackets: false
  # This option is only effective on Minecraft 1.19.1+
  # If this is true then ForceUnsignedChatPackets should also be true
  HideServerUnsignedStatus: false``` seems to fix it.. but then everyone gets these annoying prompts..

**Comment by LOOHP** [comment]
Seems like ItemsAdder is unable to read and modify unsigned player chat packets in that case. You might want to notify them about this.

**Comment by AGuyNamedJens** [comment]
Well for some reason when I set that option to false, no one seems to be able to read eachother's chats at all..

**Comment by LOOHP** [comment]
Have you tried restarting after changing that setting?

**Comment by AGuyNamedJens** [comment]
> Have you tried restarting after changing that setting?

Yes but for some reason if i disable chat signing without changing the chats themselves, everyone who locally has it enabled can't see anyone's chats anymore

**Comment by LOOHP** [comment]
If `ForceUnsignedChatPackets`is `false` then IC won't tinker with incoming chat packets. So whatever happens should be the standard behavior.

**Comment by AGuyNamedJens** [comment]
> If `ForceUnsignedChatPackets`is `false` then IC won't tinker with incoming chat packets. So whatever happens should be the standard behavior.

Yeah.. I guess it's Minecraft's behavior then..
Hopefully the bug on ItemsAdders side gets fixed quickly enough.. If i was allowed to share my JAR I could've shared the source but it's paid so, I can't:(

**Comment by LOOHP** [comment]
I do have ItemsAdder myself, but their code is extremely obfuscated so it's very painful to read and understand.

**Comment by AGuyNamedJens** [comment]
Ah bummer, I mean I get it due to the amount of illegal copies of other plugins going around the internet..
Idk if it's even possible, but wouldn't they allow to give you the source (or atleast the important part) to fix a bug?

**Comment by LOOHP** [comment]
We'll see how they respond first.

**Comment by AGuyNamedJens** [comment]
Well, I noticed the dev there returns to development in a week or 2.

I noticed that when i disabled InteractiveChat's color code handler, I can't use essentials's built-in one either? I wanted to disable it so I could use Essentials's color permissions, but I guess not..
Feels like Essentials doesn't support InteractiveChat either then?

**Comment by LOOHP** [comment]
By "disabling InteractiveChat's color handler", what did you actually do?

**Comment by AGuyNamedJens** [comment]
Chat:
  # Convert alternate color code character
  # Only one character is allowed, leave blank to disable this feature
  TranslateAltColorCode: ''
  
 set this to '' because for some reason InteractiveChat overwrites essentials's permission where only players with "essentials.chat.color" can use it (or other types of color -> https://essentialsx.net/wiki/Color-Permissions.html)

**Comment by LOOHP** [comment]
You shouldn't have to change that if you just want to not use IC's color translate on player chat messages. You can just deny players the `interactivechat.chatcolor.translate` permission

**Comment by AGuyNamedJens** [comment]
I would like to use essentials's chat coloring, it doesn't seem like i can
tell IC to give specific players specific colors

**Comment by AGuyNamedJens** [comment]
Seems like LoneDev fixed it since he's back from vacation.. No idea what the issue itself was though?
Now the part of EssentialsX.. Any idea how to get those to parse based on permissions they use, rather than allow every color from InteractiveChat?

**Comment by AGuyNamedJens** [comment]
Also ontop of that, is there an option to prevent specific messages (for example the ItemsAdder's <color text> formats from being sent to Discord's webhook?

**Comment by LOOHP** [comment]
> Seems like LoneDev fixed it since he's back from vacation.. No idea what the issue itself was though? Now the part of EssentialsX.. Any idea how to get those to parse based on permissions they use, rather than allow every color from InteractiveChat?

You can deny everybody's permission to `interactivechat.chatcolor.translate`, then IC will not touch chat colors. So other plugins such Essentials will have full control over that.

> Also ontop of that, is there an option to prevent specific messages (for example the ItemsAdder's formats from being sent to Discord's webhook?

That I feel like something you would be set in DiscordSRV instead of InteractiveChat.

**Comment by AGuyNamedJens** [comment]
> > Seems like LoneDev fixed it since he's back from vacation.. No idea what the issue itself was though? Now the part of EssentialsX.. Any idea how to get those to parse based on permissions they use, rather than allow every color from InteractiveChat?
> 
> You can deny everybody's permission to `interactivechat.chatcolor.translate`, then IC will not touch chat colors. So other plugins such Essentials will have full control over that.

Wait, that's a thing? Why can't i find IC's permissions easily.. lol

> > Also ontop of that, is there an option to prevent specific messages (for example the ItemsAdder's formats from being sent to Discord's webhook?
> 
> That I feel like something you would be set in DiscordSRV instead of InteractiveChat.
Oh right.. I thought InteractiveChat was managing that as well, considering it hides the [] templates. Will look into that :)

**Comment by AGuyNamedJens** [comment]
Yeah no i'm giving up on EssentialsX Chat, it's still not working.. Weird..

**Comment by AGuyNamedJens** [comment]
Also sometimes receiving this error: `java.lang.RuntimeException: Unable to fetch assets from "api.loohpjames.com". This could be an internet issue or "api.loohpjames.com" is down. If the plugin functions correctly after this, this error can be ignored.`
